src/bourso/client.rs

```diff
@@ -146,44 +154,161 @@ impl BoursoWebClient {
+    /// awaiting user input, and validating the authentication status. If the user is successfully
+    /// verified via the app, it returns `true`. If not, it returns `false`.
+    ///
+    /// # Returns
+    ///
+    /// A `Result` indicating whether the strong authentication was successful.
+    pub async fn handle_strong_auth_verification(&mut self) -> Result<bool> {
+        log_with_timestamp(format!("Initiating strong authentication verification.").blue());
+        let res = self
+            .client
+            .get(format!("{BASE_URL}/securisation"))
+            .headers(self.get_headers())
+            .send()
+            .await?
+            .text()
+            .await?;
+
+        self.config = extract_brs_config(&res)?;
+
+        if let Some(config) = &self.config.user_hash {
+            log_with_timestamp(format!("Retrieved user hash: `{}`", config).blue());
+        } else {
+            log_with_timestamp(format!("User hash not found in config during strong authentication.").red());
+            bail!("Strong authentication failed: User hash not found.");
+        }
+
+        let res = self
+            .client
+            .get(format!("{BASE_URL}/securisation/validation"))
+            .headers(self.get_headers())
+            .send()
+            .await?
+            .text()
+            .await?;
+
+        let token_form = extract_strong_auth_form_token(&res)?;
+        let form_state = extract_strong_auth_formstate(&res)?;
+
+        self.client
+            .request(Method::OPTIONS, format!(
+                "https://api.boursobank.com/services/api/v1.7/_user_/_{}_/session/challenge/checkwebtoapp/10305",
+                self.config.user_hash.as_mut().unwrap()
+            ))
+            .headers(self.get_headers())
+            .send()
+            .await?
+            .text()
+            .await?;
+
+        // Implement Bearer authorization to headers
+        let mut headers = self.get_headers();
+        headers.insert(
+            "authorization",
+            format!("Bearer {}", self.config.default_api_bearer)
+                .parse()
+                .unwrap(),
+        );
+
+
+        let strong_auth_url = format!(
+            "https://api.boursobank.com/services/api/v1.7/_user_/_{}_/session/challenge/startwebtoapp/10305", 
+            self.config.user_hash.as_mut().unwrap()
+        );
+
+        // Send strong auth verification to app
+        self.client.post(strong_auth_url)
+            .headers(headers.clone())
+            .body(serde_json::json!({ "form_state": form_state }).to_string())
+            .send()
+            .await?
+            .text()
+            .await?;
+
+        log_with_timestamp(format!("Strong authentication: Sending request to app").blue());
+
+        // Await for user to press enter
+        log_with_timestamp(format!("Click 'Enter' after the app verification is complete.").yellow());
```

Comment:
long lines should be splited like this
```
log_with_timestamp(
    format!(
        "Click 'Enter' after the app verification is complete."
    ).yellow()
);
```
What do you think @Tixwell ?

Reply:
This should be formatter's function. Do you want me to add rustfmt.toml and a GA to check it?

Reply:
@valeratrades sure, that would be helpful!